controllers/component_build_controller_test.go

```diff
@@ -781,98 +949,552 @@ var _ = Describe("Component initial build controller", func() {
 				Expect(d.AuthorEmail).ToNot(BeEmpty())
 				return "url", nil
 			}
-			isDeletePaCWebhookInvoked := false
-			DeletePaCWebhookFunc = func(repoUrl string, webhookUrl string) error {
-				isDeletePaCWebhookInvoked = true
-				Expect(repoUrl).To(Equal(SampleRepoLink))
+			isSetupPaCWebhookInvoked := false
+			SetupPaCWebhookFunc = func(repoUrl string, webhookUrl string, webhookSecret string) error {
+				isSetupPaCWebhookInvoked = true
 				Expect(webhookUrl).To(Equal(pacWebhookUrl))
+				Expect(webhookSecret).ToNot(BeEmpty())
+				Expect(repoUrl).To(Equal(SampleRepoLink))
 				return nil
 			}
-			isDeleteBranchInvoked := false
-			DeleteBranchFunc = func(repoUrl string, branchName string) (bool, error) {
-				isDeleteBranchInvoked = true
-				return true, nil
-			}
 
 			pacSecretData := map[string]string{"github.token": "ghp_token"}
 			createSecret(pacSecretKey, pacSecretData)
 
-			createComponentForPaCBuild(getSampleComponentData(resourceKey))
-			setComponentDevfileModel(resourceKey)
-			waitPaCFinalizerOnComponent(resourceKey)
-
-			deleteComponent(resourceKey)
+			createComponentAndProcessBuildRequest(resourceKey, BuildRequestConfigurePaCAnnotationValue)
 
+			waitSecretCreated(namespacePaCSecretKey)
+			waitSecretCreated(webhookSecretKey)
+			waitPaCRepositoryCreated(resourceKey)
 			Eventually(func() bool {
-				return isRemovePaCPullRequestInvoked
+				return isCreatePaCPullRequestInvoked
 			}, timeout, interval).Should(BeTrue())
 			Eventually(func() bool {
-				return isDeletePaCWebhookInvoked
+				return isSetupPaCWebhookInvoked
 			}, timeout, interval).Should(BeTrue())
-			Eventually(func() bool {
-				return isDeleteBranchInvoked
-			}, timeout, interval).Should(BeFalse())
 		})
 
-		It("should not block component deletion if PaC definitions removal failed", func() {
-			UndoPaCMergeRequestFunc = func(string, *gp.MergeRequestData) (webUrl string, err error) {
-				return "", fmt.Errorf("failed to create PR")
-			}
-			DeletePaCWebhookFunc = func(string, string) error {
-				return fmt.Errorf("failed to delete webhook")
+		It("should provision PaC definitions after initial build, use simple build while PaC enabled, and be able to switch back to simple build only", func() {
```

Comment:
Does it test three cases? Can we split it into three separate cases?

Reply:
It indeed tests 3 cases, but if we split it into 3 tests, then we'd need to do preparation for each step, which is duplicating the work.